### PR TITLE
Update collision for 4 0

### PR DESCRIPTION
**Pull request by kaltinril**
Added more information about how the collision code works:

1. Collision Basics
2. Detailed Explanation
3. Space Algorithms Explained (QuadTree, SpatialHash)
4. Layer
5. CollisionComponent
6. ICollisionActor

Updated the code example to work (Changed Point2 to Vector2 and Size2 to SizeF).

Added comments and notes to help beginners out.

I liked the existing code example so I kept it the same behavior.

I can make any changes suggested, fix any typos, etc, please let me know if you have any questions

https://github.com/craftworkgames/MonoGame.Extended-samples/pull/20

### CODE REVIEW

File: docs/features/collision/collision.md [hunk 1]
```diff
@@ -5,11 +5,137 @@ title: Collision
...
 ---
 
-:::warning[Not up to date]
-This page **is not** up to date for MonoGame.Extended `@mgeversion@`.  If you would like to contribute to updating this document, please [create a new PR](https://github.com/craftworkgames/craftworkgames.github.io/pulls)
+:::tip[Up to date]
+This page is **up to date** for MonoGame.Extended `@mgeversion@`.  If you find outdated information, [please open an issue](https://github.com/craftworkgames/craftworkgames.github.io/issues).
 :::
 
-## Usage
+## Collision basics
```
Comment (AristurtleDev) [inline docs/features/collision/collision.md]: Should there be a basics section and a detailed explanation section?  If possible, documents shouldn't be split between tl;dr and full.

File: docs/features/collision/collision.md [hunk 2]
```diff
@@ -5,11 +5,137 @@ title: Collision
...
+
+Think of mipmaps or approximations.  The screen is split up into N sections, and the object is either in that large section or not.
+
+Example creation of a SpatialHash:
+```csharp
+SpatialHash shash = new SpatialHash(new Vector2(32, 32));
+```
+See [Spatial Hashing](https://www.gamedev.net/tutorials/programming/general-and-gameplay-programming/spatial-hashing-r2697/) on GameDev.
+
+### Layer
```
Comment (AristurtleDev) [inline docs/features/collision/collision.md]: Wasn't sure where to put this review note so I'll do it here at the `Layer` header.  Since `Layers` are "Advanced/Optional", and the full example below doesn't make explicit use of them, it may be better to move the `Layer` section to the bottom of the document.  

For example, the first part of the document is explaining the minimum setup for collisions being the `CollisionComponent` and the `ICollisionActor` followed by the example.  Then after the example is shown, have an `Advanced` section that discusses the `Layer` object, reflect back on how the `CollisionComponent` discussed before uses the `QuadTree` by default and you can change this by creating your own layer.

So basically reorganizing the docs so it starts with the requirement pieces and then has a section on advanced use cases. The space algorithms would be moved down to that part of the documents as well.

I may be explaining this poorly, so feel free to reach out to me and we can talk more.

File: docs/features/collision/collision.md [hunk 3]
```diff
@@ -5,11 +5,137 @@ title: Collision
...
+1. CollisionComponent
+1. ICollisionActor
+
+You don't need to understand them in detail to use them.  Below are high level explanations of each.
+
+You may skip to the [Full Example](#full-example) section and ignore these details if you wish.
+
+### Space Algorithms
+
+(Advanced/Optional: You don't need to create one of these, this is just explaining how things work behind the scenes.  Skip to the [Full Example](#full-example) for the bare minimum.)
```
Comment (AristurtleDev) [inline docs/features/collision/collision.md]: Given the note concerning moving the Layer section down, this clarification wouldn't be needed anymore.  Same for the other sections that begin with clarifications on if something is required or optional.  Start with all required things before the example, then do the optional things after.

File: docs/features/collision/collision.md [hunk 4]
```diff
@@ -25,7 +151,10 @@ public interface IEntity : ICollisionActor
 
 Next, we define our entity classes
 
-The `OnCollision` method and the Bounds property come from the `ICollisionActor` interface. These will be called by the `CollisionComponent`
+The `OnCollision` method and the Bounds property come from the `ICollisionActor` interface. These will be called by the `CollisionComponent`.
+:::note[notice]
+The RandomizeVector method will give a warning/error, this will be resolved in the "Setting up the game" section when you add the Random variable to the Game1.cs file.
```
Comment (AristurtleDev) [inline docs/features/collision/collision.md]: Unless there is a specific reason someone needs to seed the `Random`, you can switch this from using `_game.Random` to instead use `Random.Shared` which will no longer need this mention of a warning/error.

**Review by AristurtleDev — CHANGES_REQUESTED** [review]
First @kaltinril, thank you again for taking the time to do these updates.  I'm sorry that I've been delayed on responding back.   I've left some requests for changes, if you have any questions, please reach out to me.

File: docs/features/collision/collision.md (hunk 1 above)
Comment (kaltinril) [reply docs/features/collision/collision.md]: I was trying to provide a high level overview since it's actually a bit complex to implement collision until you've done it once or twice.

Do you want me to remove the Section, or rename it?

File: docs/features/collision/collision.md (hunk 2 above)
Comment (kaltinril) [reply docs/features/collision/collision.md]: Gotcha, so sort of separate it into beginner/basic and advanced?

Comment (kaltinril) [reply docs/features/collision/collision.md]: @AristurtleDev  I'm still having trouble thinking of how to reorganize what I wrote up.

My main concern is that jumping straight into the example would be overwhelming for most, and I want to explain the different parts of the collision classes and how they interact with eachother.

I'm worried that leaving that out would be confusing.

Comment (kaltinril) [reply docs/features/collision/collision.md]: I'll take a stab at it and send an update

**Comment by kaltinril** [comment]
@AristurtleDev Lets try this, let me know if you want me to rearrange or move other items.

File: docs/features/collision/collision.md (hunk 1 above)
Comment (AristurtleDev) [reply docs/features/collision/collision.md]: Let's circle back around on this at a later date.  The information on collisions can be split int separate documents if needed to provide a focused page covering a specific idea such as the algorithms and layers, similar to how the Texture Handling documents are broken into three documents that build on each other to go from TextureRegion to TextureAtlas to Sprite. 

I think for now the current organization of the document should be good and we can get feedback from users and see if any questions pop up.

**Review by AristurtleDev — APPROVED** [review]
